include/swift/AST/ASTContext.h

```diff
@@ -869,6 +870,21 @@ class ASTContext final {
   /// This guarantees that resulted \p names doesn't have duplicated names.
   void getVisibleTopLevelModuleNames(SmallVectorImpl<Identifier> &names) const;
 
+  struct OverrideSignatureKey {
```

Comment:
This is an implementation detail of the ASTContext and not part of the public API, so it should be defined in ASTContext.cpp.